src/printer_tree.c

```diff
@@ -4337,3 +4418,131 @@ tree_print_parsed_submodule(struct ly_out *out, const struct lysp_submodule *sub
+    struct lyd_node *ext_data;
+    ly_bool ext_data_free;
+
+    /* get operational data with ietf-yang-library and ietf-yang-schema-mount data */
+    if ((res = lyplg_ext_get_data(ext->module->ctx, ext, (void **)&ext_data, &ext_data_free))) {
+        return res;
+    }
+
+    LY_CHECK_GOTO(res = lyd_find_xpath(ext_data, "/ietf-yang-schema-mount:schema-mounts/mount-point/shared-schema/parent-reference",
+                                       &pref_set), out);
+    if (pref_set->count == 0) {
+        goto out;
+    }
+
+    LY_CHECK_GOTO(res = ly_set_new(&results_set), out);
+
+    for (uint32_t i = 0; i < pref_set->count; ++i) {
+        struct lyd_node_term *term;
+        struct lyd_value_xpath10 *xp_val;
+        char *value;
+        struct ly_err_item *err;
+
+        term = (struct lyd_node_term *)pref_set->dnodes[i];
+        LYD_VALUE_GET(&term->value, xp_val);
+        LY_CHECK_GOTO(res = lyplg_type_print_xpath10_value(xp_val, LY_VALUE_JSON, NULL, &value, &err), out);
+        LY_CHECK_ERR_GOTO(res = lys_find_xpath(ext->module->ctx, NULL, value, 0, &snode_set), free(value), out);
+        free(value);
+        for (uint32_t sn = 0; sn < snode_set->count; sn++) {
+            struct lysc_node *top = snode_set->snodes[sn];
+
+            while (top->parent)
+                top = top->parent;
+
+            if ((res = ly_set_add(results_set, top, 0, NULL))) {
+                ly_set_free(snode_set, NULL);
+                ly_set_free(results_set, NULL);
+                goto out;
+            }
+        }
+        ly_set_free(snode_set, NULL);
+    }
+
+    *refs = results_set;
+
+out:
+    if (ext_data_free) {
+        lyd_free_all(ext_data);
+    }
+    ly_set_free(pref_set, NULL);
+    return res;
+}
+
+static LY_ERR
+trb_print_mount_point(struct trt_node *node, struct trt_wrapper wr, struct trt_printer_ctx *pc,
+                      struct trt_tree_ctx *UNUSED(tc))
+{
+    struct ly_ctx *ext_ctx;
+    const struct lys_module *mod;
+    struct trt_tree_ctx tmptc;
+    struct trt_tree_ctx *newtc;
+    struct trt_wrapper tmpwr;
+    struct ly_set *modules;
+    struct ly_set *refs = NULL;
+    LY_ERR err = LY_SUCCESS;
+
+    if (lyplg_ext_schema_mount_create_context(node->mount, &ext_ctx))
+        /* Void mount point */
+        return LY_SUCCESS;
+
+    if (ly_set_new(&modules)) {
+        err = LY_SUCCESS;
+        goto out_ctx;
+    }
+
+    trb_parent_references(node->mount, &refs);
+
+    /* build new list of modules to print.  This list will omit internal
+     * modules and modules with no nodes (e.g., iana-if-types).
+     */
+    uint32_t v = ly_ctx_internal_modules_count(ext_ctx);
+    while ((mod = ly_ctx_get_module_iter(ext_ctx, &v))) {
+        uint32_t siblings;
+
+        if ((mod->ctx->flags & LY_CTX_SET_PRIV_PARSED) && mod->compiled)
+            trm_lysc_tree_ctx(mod, pc->out, pc->max_line_length, pc, &tmptc);
+        else
+            trm_lysp_tree_ctx(mod, pc->out, pc->max_line_length, pc, &tmptc);
```

Comment:
Displaying the "mp" flag is no problem and there is enough information to figure out when `/` and `@` should be displayed when working with the parsed module, but displaying the mounted tree itself might require substantial work.  The schema_mount plugin API is dependent on `struct lysc_ext_instance`, presumably because it has module and plugin data pointers which the lysp_* version of that structure is missing.  The extension callback function (ly_ext_data_clb) also requires the lysc variant.

Maybe I have overlooked something.  What are your thoughts?

Reply:
Yes the schema_mount plugin API needs a compiled module. If there is a module that is only parsed, I would just print the `mp` flag for now and not do any additional compilation. So if the mounted module is only parsed, its mount points are not resolved (only the `mp` flags are printed).
I will try to summarize the situation:
1. The compiled module (lysc) has a mount point:
1.1 The mounted module is compiled and it can be printed.
1.2 The mounted module is only parsed but it can be printed (_not currently implemented in PR_ and we agreed that `mp`, `@`, `/` symbols can be printed.).
2. If the parsed module (lysp) has a mount point:
2.1 The mounted module cannot be printed.

Reply:
I think I have accounted for all of these.  Scenario 1.2 requires some manual testing because, as it is, the schema mount plugin will always create a context for the mounted modules so that they are compiled.  I tried this out with a temporary change
```
--- a/src/printer_tree.c
+++ b/src/printer_tree.c
@@ -4469,6 +4469,9 @@ trb_print_mount_point(struct trt_node *node, struct trt_wrapper wr, struct trt_p
         return LY_SUCCESS;
     }
 
+    if ((err = ly_ctx_new(".", 0, &ext_ctx)))
+        return err;
+
     if (lyplg_ext_schema_mount_create_context(node->mount, &ext_ctx)) {
         /* Void mount point */
         return LY_SUCCESS;
```
to force the plugin to start with an existing (empty) context.

Reply:
When I tried this temporary change, I didn't see any result. Nevermind. You can test scenario 1.2 by adding to `sm-context-extension.xml` something like:
```
     <import-only-module>
       <name>sm-mod</name>
       <revision>2017-01-26</revision>
       <namespace>urn:yanglint:sm-mod</namespace>
     </import-only-module>
```
Also add `import sm-mod {...` to the `sm-extension` module. And also of course you need to add a module `sm-mod.yang` with some nodes. The parsed module print should be visible using your command:
`../../../build/yanglint -f tree -p ../../../models -p . -Y ./sm-context-main.xml -x ./sm-context-extension.xml sm-main.yang`
When I tried such an example, the `printer_tree` failed. Please check it out.

Reply:
I can reproduce the error and will take a look at it.

Reply:
I just pushed an update to address this.